### PR TITLE
MOSS Results Viewer

### PR DESCRIPTION
**Pull request by khalilsarwari**
Looking forward to some feedback!

It's parallel!
screen shot 2017-11-15 at 7 59 02 pm

### CODE REVIEW

File: server/controllers/admin.py [hunk 1]
```diff
@@ -925,6 +927,59 @@ def assignment_moss_results(cid, aid):
...
+    for file in primary_files:
+        list_of_lines = primary_files[file].splitlines()
+        unique_lines = copy.deepcopy(list_of_lines)
+        for primary_matches in moss_result.primary_matches[file]:
+            for line in range(primary_matches[0], primary_matches[1] + 1):
+                unique_lines[line] = -1
+        primary_unmatch[file] = ''.join([line + '\n' for line in unique_lines if line != -1])
+    primary_hlt = highlight.diff_files(primary_files, primary_unmatch, diff_type)
+
+    secondary_files = moss_result.secondary.files()
```
Comment (khalilsarwari) [inline server/controllers/admin.py]: Make helper function to get the hlts.

**Comment by epai** [comment]
This is awesome!  Padding is a bit off.  I can take a look at it this weekend.  @jathak what do you think of this moss viewer?

**Comment by jathak** [comment]
I'll try to do a review after my section tomorrow. Initial impressions look
very good. Great job, Khalil!

**Comment by epai** [comment]
Looks clean!  I'll stage it on ok-staging.cs61a.org, and if it looks good we can merge

**Comment by epai** [comment]
I'm going to add a few changes:
- Use backup url instead of the student url
- rename "primary group" and "secondary group" to just "group" (to save space)
- make file boxes collapsable (big change)
- make _files_ (not sections) vertically align (big change)

**Comment by jathak** [comment]
It looks like the Moss result page is missing a title

**Comment by jathak** [comment]
Trying on staging, this works great with Ants, but I'm getting a server error when trying to load the page for Scheme.

**Comment by khalilsarwari** [comment]
Hmm that's a bit strange. Did you run the MOSS jobs for scheme? Do those complete? Also what is the staging site for ok? I'd like to take a look if possible/needed.

**Comment by jathak** [comment]
Just re-ran Moss after adding the templates. Some results work; others still error. Sending you the link in Slack

**Comment by epai** [comment]
Alright, I finished the changes I proposed.

snip20171118_13

**Comment by epai** [comment]
But yeah, I'm getting that some results error as well.  Can you share the photo to #devops @jathak?

**Pull request merged**